templates/commands/render/render_test.go

```diff
@@ -719,26 +708,33 @@ steps:
 				t.Errorf("dest directory contents were not as expected (-got,+want): %s", diff)
 			}
 
-			gotBackupContents := common.LoadDirWithoutMode(t, backupDir)
-			gotBackupContents = stripFirstPathElem(gotBackupContents)
+			var gotBackupContents map[string]string
+			backupSubdir, ok := mustGlob(t, filepath.Join(backupDir, "*")) // When a backup directory is created, an unpredictable timestamp is added, hence the "*"
+			if ok {
+				gotBackupContents = common.LoadDirWithoutMode(t, backupSubdir)
+			}
 			if diff := cmp.Diff(gotBackupContents, tc.wantBackupContents, cmpopts.EquateEmpty()); diff != "" {
 				t.Errorf("backups directory contents were not as expected (-got,+want): %s", diff)
 			}
 		})
 	}
 }
 
-// Since os.MkdirTemp adds an extra random token, we strip it back out to get
-// determistic results. Input map keys should use forward slash separator.
-func stripFirstPathElem(m map[string]string) map[string]string {
-	out := map[string]string{}
-	for k, v := range m {
-		// Panic in the case where k has no slashes; this is just a test helper.
-		elems := strings.Split(k, "/")
-		newKey := path.Join(elems[1:]...)
-		out[newKey] = v
+func mustGlob(t *testing.T, glob string) (string, bool) {
+	t.Helper()
+
+	matches, err := filepath.Glob(glob)
+	if err != nil {
+		t.Fatalf("couldn't find template directory: %v", err)
 	}
-	return out
+	switch len(matches) {
+	case 0:
+		return "", false
+	case 1:
+		return matches[0], true
+	}
+	t.Fatalf("got %d matches for glob %q, wanted 1: %s", len(matches), glob, matches)
+	panic("unreachable")
```

Comment:
How would we get here?

Reply:
Some background here: this is used to find one directory within a temp directory contaning several directories, while ignoring a random cookie at the end of the name. For example, suppose there's a temp directory containing two subdirectories, `template-copy-192837` and `scratch-977245`. If you want to find the template directory, you pass a glob of `template-copy-*` and you'd find `template-copy-192837`.

So the Fatalf covers the case where the caller's glob wasn't specific enough to match a single directory. This could happen if the glob was just `*` or `*-*`.

Reply:
Sorry, I'm not questioning the fatalf. I'm questioning the panic after it. Fatal should stop the test immediately - I'm not clear how we'd ever hit L737 - it's unreachable code.

Reply:
Same question

Reply:
Ah, I see. The compiler complains about a missing return statement because the compiler doesn't know that t.Fatalf() never returns. The panic is the standard way of handling that.

Reply:
Added a comment.